### PR TITLE
Fix id2label assignment in run_classification.py

### LINKED ISSUE #28589: Fix id2label assignment in run_classification.py
### System Info

- `transformers` version: 4.36.2
- Platform: macOS-13.4-arm64-arm-64bit
- Python version: 3.10.10
- Huggingface_hub version: 0.20.2
- Safetensors version: 0.4.0
- Accelerate version: 0.23.0
- Accelerate config:    not found
- PyTorch version (GPU?): 2.1.0 (False)
- Tensorflow version (GPU?): not installed (NA)
- Flax version (CPU?/GPU?/TPU?): not installed (NA)
- Jax version: not installed
- JaxLib version: not installed
- Using GPU in script?: no
- Using distributed or parallel set-up in script?: no

### Who can help?

@ArthurZucker @younesbelkada 

### Information

- [X] The official example scripts
- [ ] My own modified scripts

### Tasks

- [X] An officially supported task in the `examples` folder (such as GLUE/SQuAD, ...)
- [ ] My own task or dataset (give details below)

### Reproduction

```
export TASK_NAME=mrpc

python examples/pytorch/text-classification/run_glue.py \
  --model_name_or_path bert-base-cased \
  --task_name $TASK_NAME \
  --do_train \
  --do_eval \
  --max_seq_length 128 \
  --per_device_train_batch_size 32 \
  --learning_rate 2e-5 \
  --num_train_epochs 3 \
  --output_dir /tmp/$TASK_NAME/
  ```

### Expected behavior

**Issue Description:**
The `run_classification.py` script currently has an issue where the assignment of `id2label` in the model's config is incorrect. The problem arises from copying `config.label2id` without modifying it later on. This issue was introduced with transformers version 4.36.

**Steps to Reproduce:**
1. Execute the `run_classification.py` script with a configuration file.
2. Inspect the `id2label` attribute in the model's config.

**Expected Behavior:**
The `id2label` attribute should be assigned correctly, reflecting the label-to-id mapping.

**Actual Behavior:**
The `id2label` attribute is assigned based on the original `config.label2id`, leading to incorrect mapping.

**Proposed Solution:**
Modify the following line in `run_classification.py`:

```python
model.config.id2label = {id: label for label, id in config.label2id.items()}
```

to:

```python
model.config.id2label = {id: label for label, id in label_to_id.items()}
```

This change ensures that the correct mapping is used.

### PR DESCRIPTION
**Pull request by jheitmann**
# What does this PR do?

This pull request addresses an issue in the `run_classification.py` script where the assignment of the `id2label` attribute in the model's config is incorrect. The current implementation copies `config.label2id` without modifying it, leading to an incorrect mapping. The proposed fix ensures that the `id2label` attribute is assigned based on the correct mapping (`label_to_id`) to resolve this issue.

**Changes Made:**
- Modified the assignment of `id2label` in the `run_classification.py` script to use the correct label-to-id mapping.

**Context:**
This issue was introduced with transformers version 4.36, and the incorrect assignment can lead to unexpected behavior in the script.

Fixes #28589 

## Before submitting
- [ ] This PR fixes a typo or improves the docs (you can dismiss the other checks if that's the case).
- [X] Did you read the [contributor guideline](https://github.com/huggingface/transformers/blob/main/CONTRIBUTING.md#start-contributing-pull-requests),
      Pull Request section?
- [ ] Was this discussed/approved via a Github issue or the [forum](https://discuss.huggingface.co/)? Please add a link
      to it if that's the case.
- [ ] Did you make sure to update the documentation with your changes? Here are the
      [documentation guidelines](https://github.com/huggingface/transformers/tree/main/docs), and
      [here are tips on formatting docstrings](https://github.com/huggingface/transformers/tree/main/docs#writing-source-documentation).
- [ ] Did you write any new necessary tests?

## Who can review?

- @ArthurZucker  
- @younesbelkada

### CODE REVIEW

**Review by amyeroberts — APPROVED** [review]
Thanks for fixing!
